### PR TITLE
fall back through providers if price tiers are not available (bug 1136212)

### PR DESCRIPTION
**Pull request by andymckay**

### CODE REVIEW

File: webpay/pay/tasks.py [hunk 1]
```diff
@@ -177,21 +176,33 @@ def get_provider_seller_uuid(issuer_key, product_data, provider_names):
...
+                         s='found' if provider_seller_uuid else 'NOT FOUND'))
+
+        if provider_seller_uuid:
+            prices = mkt_client.get_price(price_point, provider=provider)
+            if not prices['prices']:
+                log.info('No prices for provider: {p}'.format(p=provider))
+                continue
+
+            log.info('Price found for provider: {p}' .format(p=provider))
+            return ProviderHelper(provider), provider_seller_uuid, prices
```
Comment (kumar303) [inline webpay/pay/tasks.py]: :+1: for all this logging

File: webpay/pay/tasks.py [hunk 2]
```diff
@@ -246,12 +255,14 @@ def start_pay(transaction_uuid, notes, user_uuid, provider_names, **kw):
 
         # Ask the marketplace for a valid price point.
         # Note: the get_price_country API might be more helpful.
```
Comment (kumar303) [inline webpay/pay/tasks.py]: are thee comments still applicable?

File: webpay/pay/tests/test_tasks.py [hunk 3]
```diff
@@ -577,6 +577,46 @@ def test_bango_used_when_boku_supported_but_not_setup_in_solitude(self):
...
+                'bango': self.bango_seller_uuid,
+                'boku': self.boku_seller_uuid,
+            },
+            'seller': '/generic/seller/1',
+            'external_id': 'external_id',
+        }
+
+        def no_boku_price(pricePoint, provider):
+            if provider == 'boku':
+                return {'prices': []}
```
Comment (kumar303) [inline webpay/pay/tests/test_tasks.py]: does the API return an empty list or None?

File: webpay/pay/tests/test_tasks.py [hunk 4]
```diff
@@ -577,6 +577,46 @@ def test_bango_used_when_boku_supported_but_not_setup_in_solitude(self):
...
+        self.mkt.webpay.prices.return_value = pricey
+
+        tasks.start_pay(self.transaction_uuid, self.notes, self.user_uuid,
+                        [p.name for p in self.providers])
+
+        self.solitude.bango.product.get_object_or_404.assert_called_with(
+            seller_product__seller=1,
+            seller_product__external_id='external_id'
+        )
+        ok_(self.mkt.webpay.prices.call_count, 2)
```
Comment (kumar303) [inline webpay/pay/tests/test_tasks.py]: could this also assert a call to ``self.solitude.bango.billing.post`` ? That seems like the important one.

File: webpay/pay/tests/test_tasks.py [hunk 5]
```diff
@@ -796,6 +836,61 @@ def test_no_error_type(self):
...
+
+    def test_no_price(self, get_price):
+        get_price.return_value = {'prices': []}
+
+        with self.assertRaises(NoValidSeller):
+            tasks.get_best_provider(**self.default())
+        eq_(get_price.call_args_list,
+            [call('10', provider='bango'),
+             call('10', provider='boku'),
+             call('10', provider='reference')])
```
Comment (kumar303) [inline webpay/pay/tests/test_tasks.py]: nice

**Comment by kumar303** [comment]
I see you removed public_id from the provider lookup but maybe that was just for logging, I forget.

r+wc

File: webpay/pay/tasks.py (hunk 2 above)
Comment (andymckay) [inline webpay/pay/tasks.py]: nope, lets kill 'em

File: webpay/pay/tests/test_tasks.py (hunk 3 above)
Comment (andymckay) [inline webpay/pay/tests/test_tasks.py]: empty list

File: webpay/pay/tests/test_tasks.py (hunk 4 above)
Comment (andymckay) [inline webpay/pay/tests/test_tasks.py]: that seems like a good idea

**Comment by andymckay** [comment]
Yeah public_id was just logging.

**Pull request merged**